File: src/main/java/io/dustin/finstage/financial/domain/Symbol.java
package io.dustin.finstage.financial.domain;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@Getter
@EqualsAndHashCode
@ToString
public class Symbol {
    private final String symbol;
    private final String name;
    private final String country;

    private Symbol(String symbol, String name, String country) {
        this.symbol = symbol;
        this.name = name;
        this.country = country;
    }

    public static Symbol of(String symbol, String name, String country) {
        return new Symbol(symbol, name, country);
    }
}

File: src/main/kotlin/io/dustin/finstage/financial/domain/Symbol.kt
package io.dustin.finstage.financial.domain

data class Symbol(
        val symbol: String,
        val name: String,
        val country: String
) {
    companion object {
        fun of(symbol: String, name: String, country: String): Symbol {
            return Symbol(symbol, name, country)
        }
    }
}
